exonum-java-binding/core/rust/src/proxy/runtime.rs

```diff
@@ -0,0 +1,624 @@
+            Some(ArtifactProtobufSpec::default())
+        } else {
+            None
+        }
+    }
+
+    fn start_service(&mut self, spec: &InstanceSpec) -> Result<(), ExecutionError> {
+        let adapter = self.runtime_adapter.as_obj().clone();
+        let service_name = spec.name.clone();
+        let id = spec.id;
+        let artifact = self.parse_artifact(&spec.artifact)?;
+
+        let instance = Instance::new(spec.id, spec.name.clone());
+        self.can_start_service(&instance)?;
+
+        Self::parse_jni(self.exec.with_attached(|env| {
+            let name = JObject::from(env.new_string(service_name)?);
+            let artifact_id = JObject::from(env.new_string(artifact.to_string())?);
+
+            env.call_method_unchecked(
+                adapter,
+                runtime_adapter::create_service_id(),
+                JavaType::Primitive(Primitive::Void),
+                &[
+                    JValue::from(name),
+                    JValue::from(id as i32),
+                    JValue::from(artifact_id),
+                ],
+            )
+            .map(|_| ())
+        }))?;
+
+        self.add_started_service(instance);
+        Ok(())
+    }
+
+    fn configure_service(
+        &self,
+        fork: &Fork,
+        descriptor: InstanceDescriptor,
+        parameters: Any,
+    ) -> Result<(), ExecutionError> {
+
+        Self::parse_jni(self.exec.with_attached(|env| {
+            let id = descriptor.id as i32;
+            let view_handle = to_handle(View::from_ref_fork(fork));
+            let params = JObject::from(env.byte_array_from_slice(&parameters.into_bytes())?);
+
+            env.call_method_unchecked(
+                self.runtime_adapter.as_obj(),
+                runtime_adapter::configure_service_id(),
+                JavaType::Primitive(Primitive::Void),
+                &[
+                    JValue::from(id),
+                    JValue::from(view_handle),
+                    JValue::from(params),
+                ],
+            )?;
+
+            Ok(())
+        }))
+    }
+
+    fn stop_service(&mut self, descriptor: InstanceDescriptor) -> Result<(), ExecutionError> {
+        let adapter = self.runtime_adapter.as_obj();
+
+        Self::parse_jni(self.exec.with_attached(|env| {
+            let id = descriptor.id as i32;
+
+            env.call_method_unchecked(
+                adapter,
+                runtime_adapter::stop_service_id(),
+                JavaType::Primitive(Primitive::Void),
+                &[
+                    JValue::from(id),
+                ],
+            )
+            .map(|_| ())
+        }))?;
+
+        self.remove_started_service(&descriptor.id);
+        Ok(())
+    }
+
+    fn execute(
+        &self,
+        _dispatcher: &Dispatcher,
+        context: &mut ExecutionContext,
+        call_info: CallInfo,
+        arguments: &[u8],
+    ) -> Result<(), ExecutionError> {
+        let tx = if let (Some(key), Some(hash)) =
```

Comment:
Is there an easier way to check that? If not, I'd request one from core, because this way is not intuitive.